testutils/fake/store.go

```diff
@@ -64,7 +64,10 @@ func NewCachedMultiStore(ms MultiStore) sdk.CacheMultiStore {
 		if store, ok := cached.kvstore[storeKey.String()]; ok {
 			return store
 		}
-		store := NewTestKVStore()
+		store := newTestKVStore()
+		store.write = func() {
+			ms.kvstore[storeKey.String()] = store
+		}
```

Comment:
cached fake stores didn't work properly when the store didn't exist before the caching